lib/hashie/dash.rb

```diff
@@ -174,10 +174,15 @@ def update_attributes!(attributes)
 
     private
 
-    def initialize_attributes(attributes)
+    def initialize_attributes(attributes, options = {})
+      return unless attributes
+
+      set_nils = options.delete(:set_nils)
+
       attributes.each_pair do |att, value|
+        next if value.nil? && !set_nils
         self[att] = value
-      end if attributes
+      end
     end
```

Comment:
I think you should break up this function instead in two and kill the option, since it's used very differently in different places.